Basic Structure, no logic

VbKoshak opened the pull request.

VbKoshak committed:
core-3: add basic route classes

VbKoshak committed:
core-3: minor road fixes

VbKoshak committed:
core-3: add car implementation

VbKoshak committed:
core-3: add major game classes

VbKoshak committed:
core-3: add Support class + minor fixes

fix: lexical mistakes
add:  some explanation comments
feat: no more surface-like classes, only objects
feat: no influence in surface

VbKoshak committed:
core-3: put some basic logic + fixed

fix: isEmpty check was wrong and not necessary at all
docs: add some more comments
feat: add some basic  logic and race drawing

shukal94 commented on src/edu/solvd/mentoring/Garage.java:
```diff
@@ -0,0 +1,29 @@
+    return stock;
+  }
+
+  public void addCarDetail(CarDetail detail, Integer price){
```
let's make all these methods as static

shukal94 reviewed (commented):
let's go on with logic

VbKoshak committed:
core-3: race logic implemetation

feat: add some supporting methods
feat: add Participant class
fix: major Race logic change

VbKoshak commented:
--DeadLine--

VbKoshak committed:
core-3: implement angle functionality

add speed decrease by 40% if angle is bigger than 60*

VbKoshak committed:
core-3: add static methods

changed Support methods to static

VbKoshak committed:
core-3: add generatots

fix: minor bugs in types
feat: add generators for:
    CarBase
    Engine
    Wheel
    Car
    Route
feat: add private methods to sup generators

VbKoshak committed:
core-3: clean code

shukal94 commented on src/edu/solvd/mentoring/Support.java:
```diff
@@ -0,0 +1,417 @@
+  }
+
+  private static void initWheels(){
+      HashMap<Integer,Short> wheelClutch1 = new HashMap<>();
```
Map<Integer,Short> wheelClutch1 = new HashMap<>()

VbKoshak replied:
Is there any difference?

shukal94 commented on src/edu/solvd/mentoring/Support.java:
```diff
@@ -0,0 +1,417 @@
+    avg = Math.round(avg/clutch.size());
+    DetailQuality dq;
+    switch (avg) {
+      case 1:
```
why is it empty?

VbKoshak replied:
When Support.getSmth(id) calles first time nothing is initialized, soin order not to initialized unnecessary parts, like wheels when we try to get CarBase and not spend time

VbKoshak replied:
Not sure about java, but
 case 3:
 case 2: 
    a = 2;
    break;
Means that in case 2 and 3 a Will take 2, code goes until the first break

shukal94 replied:
actually, usage of switch-case is rare, so it should be tested in java, not sure about this. 
I suppose to make DetailQuality enum which can pass GOOD(1), MEDIUM(0.8), BAD(0.7) and so on and just multiply that coefficient to your detail chars. So you will avoid switch and could make any number of qualities you like without additional code changes

VbKoshak replied:
I actually thougth in reverse way, that detail quality depends on total characteristicks it has

shukal94 commented on src/edu/solvd/mentoring/Support.java:
```diff
@@ -0,0 +1,417 @@
+  }
+
+  public static Route getGeneratedRoute(int stagesCount, HashMap<Integer,Surface> surfaces) {
+    final int DEFAULT_VECTOR_COUNT = 3;
```
lets move this into class level (as static)

shukal94 reviewed (approved):
see comments

VbKoshak committed:
fix: correct distance calc

VbKoshak committed:
feat: toString for Point

VbKoshak committed:
fix: nullPointer fix initGarage

VbKoshak committed:
feat: made Garage static class

Garage getGarage() is now replaced by static void initGarage()
all Garage methods are now static

VbKoshak committed:
fix: made 1 support constant for names

defaultName -> DEFAULT_NAME

VbKoshak committed:
fix: HashMap -> Map

VbKoshak committed:
fix: move final vars in Class scope

MAX_INCREASE -> MAX_AXISLENGTH_INCREASE
DEFAULT_VECTOR_COUNT -> DEFAULT_ROUTES_VECTOR_COUNT

VbKoshak committed:
fix: detailQulity -> detailRarity

renamed, so it is more to its meaning
calculating detailRarity is now not user's reposibility

VbKoshak committed:
feat: add Costants class to save all final vars

VbKoshak committed:
doc: add javadoc

VbKoshak committed:
doc: finished docs, Constants -> Settings

VbKoshak committed:
fix: changeg Garage#getPrice return type to int

VbKoshak committed:
fix: copying constructor Point -> Point#clone

VbKoshak committed:
fix: changed to more abstract types

VbKoshak committed:
feat: dd coping constructor for Point

VbKoshak committed:
Merge pull request #2 from VbKoshak/docs

doc: add javadoc
